java/statement/builder/StatementTypeBuilder.java

```diff
@@ -46,72 +47,67 @@ default StatementType type(Graql.Token.Type type) {
         return type(type.toString());
     }
     
-    /**
-     * @param name a string that this variable's label must match
-     * @return this
-     */
     @CheckReturnValue
-    default StatementType type(String name) {
-        return type(new TypeProperty(name));
+    default StatementType type(String label) {
+        return type(new TypeProperty(Label.of(label, null)));
+    }
+
+    @CheckReturnValue
+    default StatementType type(String label, String scope) {
+        return type(new TypeProperty(Label.of(label, scope)));
+    }
+
+    @CheckReturnValue
+    default StatementType type(Label label) {
+        return type(new TypeProperty(label));
     }
 
-    /**
-     * set this concept type variable as abstract, meaning it cannot have direct instances
-     *
-     * @return this
-     */
     @CheckReturnValue
     default StatementType isAbstract() {
         return type(AbstractProperty.get());
     }
 
     default StatementType sub(Graql.Token.Type type) {
-        return sub(type.toString());
+        return sub(type.toString(), null);
+    }
+
+    @CheckReturnValue
+    default StatementType sub(String type, @Nullable String scope) {
+        return sub(Graql.type(type, scope));
     }
 
-    /**
-     * @param type a concept type id that this variable must be a kind of
-     * @return this
-     */
     @CheckReturnValue
     default StatementType sub(String type) {
         return sub(Graql.type(type));
     }
 
-    /**
-     * @param type a concept type that this variable must be a kind of
-     * @return this
-     */
     @CheckReturnValue
     default StatementType sub(Statement type) {
         return sub(new SubProperty(type));
     }
 
+    @CheckReturnValue
+    default StatementType sub(SubProperty property) {
+        return type(property);
+    }
+
     default StatementType subX(Graql.Token.Type type) {
-        return subX(type.toString());
+        return subX(type.toString(), null);
     }
 
-    /**
-     * @param type a concept type id that this variable must be a kind of, without looking at parent types
-     * @return this
-     */
     @CheckReturnValue
-    default StatementType subX(String type) {
-        return subX(Graql.type(type));
+    default StatementType subX(String type, @Nullable String scope) {
```

Comment:
This also doesn't need `@Nullable`. The user of this class needs to decide which method to call if they have or not have `scope`.